unliftio/src/UnliftIO/Exception.hs

```diff
@@ -422,7 +422,7 @@ withException thing after = withRunInIO $ \run -> EUnsafe.uninterruptibleMask $
     case res1 of
         Left e1 -> do
             -- see explanation in bracket
-            _ :: Either SomeException b <- EUnsafe.try $ run $ after e1
+            _ :: Either SomeException b <- EUnsafe.try $ restore $ run $ after e1
```

Comment:
So, this line fixes the test. But it also removes the guarantee that ``action `onException` cleanup`` will definitely complete `cleanup`.